src/nodejs/hooks/seeker_agent.go

```diff
@@ -0,0 +1,389 @@
+	}
+
+	err, seekerLibraryToInstall := h.downloadAgent(compiler)
+	if err != nil {
+		return err
+	}
+
+	h.Log.Info("Before Installing seeker agent dependency")
+	err = h.updateNodeModules(seekerLibraryToInstall, compiler.BuildDir())
+	if err != nil {
+		return err
+	}
+	h.Log.Info("After Installing seeker agent dependency")
+	err = h.createSeekerEnvironmentScript(compiler)
+	if err != nil {
+		return errors.New("Error creating seeker-env.sh script: " + err.Error())
+	}
+	h.Log.Info("Done creating seeker-env.sh script")
+	return nil
+}
+
+func (h SeekerAfterCompileHook) prependRequire(compiler *libbuildpack.Stager) error {
+	entryPointPath := os.Getenv(EntryPointFile)
+	if entryPointPath == "" {
+		return nil
+	}
+	h.Log.Debug("Adding Seeker agent require to application entry point %s", entryPointPath)
+	return h.addSeekerAgentRequire(compiler.BuildDir(), entryPointPath)
+}
+
+func (h SeekerAfterCompileHook) addSeekerAgentRequire(buildDir string, pathToEntryPointFile string) error {
+	absolutePathToEntryPoint := filepath.Join(buildDir, pathToEntryPointFile)
+	h.Log.Debug("Trying to prepend %s to %s", SeekerRequire, absolutePathToEntryPoint)
+	c, err := ioutil.ReadFile(absolutePathToEntryPoint)
+	if err != nil {
+		return err
+	}
+	// do not require twice
+	if pattern.Match(c) {
+		return nil
+	}
+	return ioutil.WriteFile(absolutePathToEntryPoint, append([]byte(SeekerRequire), c...), 0644)
+}
+
+func assertServiceCredentialsValid(credentials SeekerCredentials) error {
+	errorFormat := "mandatory `%s` is missing in Seeker service configuration"
+	if credentials.SeekerServerURL == "" {
+		return fmt.Errorf(errorFormat, "seeker_server_url")
+	}
+	return nil
+}
+
+func (h SeekerAfterCompileHook) downloadAgent(compiler *libbuildpack.Stager) (error, string) {
+	parsedEnterpriseServerUrl, err := url.Parse(h.serviceCredentials.SeekerServerURL)
+	if err != nil {
+		return err, ""
+	}
+	parsedEnterpriseServerUrl.Path = path.Join(parsedEnterpriseServerUrl.Path, agentDownloadPath)
+	agentDownloadAbsoluteUrl := parsedEnterpriseServerUrl.String()
+	h.Log.Info("Agent download url %s", agentDownloadAbsoluteUrl)
+	var seekerTempFolder = filepath.Join(os.TempDir(), "seeker_tmp")
+	seekerLibraryPath := filepath.Join(os.TempDir(), "seeker-agent.tgz")
+	os.RemoveAll(seekerTempFolder)
+	os.Remove(seekerLibraryPath)
+	err = os.MkdirAll(seekerTempFolder, 0755)
+	if err != nil {
+		return err, ""
+	}
+	agentZipAbsolutePath := path.Join(seekerTempFolder, "seeker-node-agent.zip")
+	h.Log.Info("Downloading '%s' to '%s'", agentDownloadAbsoluteUrl, agentZipAbsolutePath)
+	if err = h.Downloader.DownloadFile(agentDownloadAbsoluteUrl, agentZipAbsolutePath); err != nil {
+		return err, ""
+	}
+	// no native zip support for unzip - using shell utility
+	err = h.Unzzipper.Unzip(agentZipAbsolutePath, os.TempDir())
+	if err != nil {
+		return err, ""
+	}
+	if _, err := os.Stat(seekerLibraryPath); os.IsNotExist(err) {
+		return errors.New("Could not find " + seekerLibraryPath), ""
+	}
```

Comment:
We also have this factored out in the same lib as above:
https://github.com/cloudfoundry/libbuildpack/blob/4333d93581410923d3f7f9b8daa539f4a3c23aa1/util.go#L226

Reply:
fixed